### PR TITLE
Expose auto in dataclass

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Exposes auto in `DeepSpeedPlugin` docs and changes the default to `auto` (@pacman100 let me know if we should still keep it as `None`)

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@pacman100

### CODE REVIEW

File: src/accelerate/utils/dataclasses.py [hunk 1]
```diff
@@ -495,7 +495,10 @@ class DeepSpeedPlugin:
         },
     )
     gradient_accumulation_steps: int = field(
-        default=None, metadata={"help": "Number of steps to accumulate gradients before updating optimizer states"}
+        default="auto",
```
Comment (pacman100) [inline src/accelerate/utils/dataclasses.py:498]: Hello Zach, Thank you for adding this, but if they specify it via accelerate config/cmd arg, then it won't be updated as the current logic is to check the env variable if this attribute is `None`

Comment (muellerzr) [reply src/accelerate/utils/dataclasses.py:498]: Will keep as none then! :)

Comment (muellerzr) [reply src/accelerate/utils/dataclasses.py]: (And modify the string as such)

**Review by pacman100 — APPROVED** [review]
Thank you @muellerzr, LGTM!

**Pull request merged**